src/parser.ts

```diff
@@ -3130,18 +3132,15 @@ class Parser {
                 } else {
                     hasConstructor.value = true;
                 }
-                property.kind = 'constructor';
+                kind = 'constructor';
             }
         } else {
-            if (!property.computed && (property.key.name || property.key.value.toString()) === 'prototype') {
+            if (!computed && this.isPropertyKey(key, 'prototype')) {
```

Comment:
is there any test you can add that checks the `(key.type === Syntax.Literal && key.value === value)` part of `this.isProperty` for this condition?